engine/lib/phx/script/ffi_gen/Engine.lua

```diff
@@ -17,29 +17,52 @@ function Loader.defineType()
 
     do -- C Definitions
         ffi.cdef [[
-            void       Engine_Free       (Engine*);
-            Window*    Engine_Window     (Engine*);
-            Input*     Engine_Input      (Engine*);
-            EventBus*  Engine_EventBus   (Engine*);
-            TaskQueue* Engine_TaskQueue  (Engine*);
-            HmGui*     Engine_HmGui      (Engine*);
-            void       Engine_Abort      ();
-            int        Engine_GetBits    ();
-            double     Engine_GetTime    (Engine const*);
-            cstr       Engine_GetVersion ();
-            void       Engine_Exit       (Engine*);
-            void       Engine_Terminate  ();
-            void       Engine_Update     ();
+            void       Engine_Free                               (Engine*);
+            Window*    Engine_Window                             (Engine*);
+            Input*     Engine_Input                              (Engine*);
+            EventBus*  Engine_EventBus                           (Engine*);
+            TaskQueue* Engine_TaskQueue                          (Engine*);
+            HmGui*     Engine_HmGui                              (Engine*);
+            void       Engine_Abort                              ();
+            int        Engine_GetBits                            ();
+            double     Engine_GetTime                            (Engine const*);
+            cstr       Engine_GetVersion                         ();
+            void       Engine_Exit                               (Engine*);
+            void       Engine_Terminate                          ();
+            void       Engine_Update                             ();
+            bool       Engine_StartRenderThread                  (Engine*);
```

Comment:
I'd suggest to create a separate RenderThread struct and not mix it into the engine.
The same way as we did with EventBus, TaskQueue and HmGui.